pages/api/data/chunks/[upload].js

```diff
@@ -0,0 +1,88 @@
+import fs from "fs";
+import { promisify } from "util";
+import Upload from "~/node_common/chunkUpload";
+import * as Data from "~/node_common/data";
+import * as LibraryManager from "~/node_common/managers/library";
+import * as MW from "~/node_common/middleware";
+import * as Utilities from "~/node_common/utilities";
+
+const initCORS = MW.init(MW.CORS);
+const initAuth = MW.init(MW.RequireCookieAuthentication);
+
+// NOTE(jim): To support multipart request.
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+export default async (req, res) => {
+  initCORS(req, res);
+  initAuth(req, res);
+
+  try {
+    const assembleChunks = await new Upload({ req, maxFileSize: 200, maxChunkSize: 10 }).upload();
+
+    if (!assembleChunks) {
+      return res.status(201).send({
+        decorator: "SERVER_CHUNK_UPLOADED",
+        data: {},
+      });
+    }
+
+    // Note(Aminejvm) Handling the request after receiving the last chunk.
+    const { mime, filePath } = await assembleChunks();
+    const file = await promisify(fs.readFile)(filePath);
```

Comment:
If I understand this correctly, you're still storing the file locally here?

Reply:
Indeed it does.